Makes/ViewController.m

```diff
@@ -0,0 +1,146 @@
+//
+//  ViewController.m
+//  Makes
+//
+//  Created by Mohamed Ragab on 1/8/17.
+//  Copyright © 2017 Mohamed Ragab. All rights reserved.
+//
+
+#import "ViewController.h"
+#import "Make.h"
+#import "AFNetworking.h"
+#import "Masonry.h"
+#import "MakesCollectionViewCell.h"
+#import "MakeDetailViewController.h"
+
+
+
+
+@interface ViewController ()
+@property (nonatomic, strong) NSMutableArray *makes;
+@property (nonatomic,strong) NSCache* imagesCache;
+@end
+
+@implementation ViewController
+
+- (void)loadMakes
+{
+    AFHTTPSessionManager *manager = [AFHTTPSessionManager manager];
+    
+    NSString *path = [[NSBundle mainBundle] pathForResource:@"url_string_file" ofType:@"txt"];
+    NSString *content = [NSString stringWithContentsOfFile:path encoding:NSUTF8StringEncoding error:nil];
+
+    
+    [manager GET:content parameters:nil progress:nil success:^(NSURLSessionTask *task, id responseObject) {
+        
+        NSArray* dictionariesArray = (NSArray*)responseObject;
+        NSLog(@"%@",responseObject);
+        
+        Make* makeHolder = [[Make alloc]init];
+        NSError * parserError;
+        NSMutableArray* indexPaths = [NSMutableArray array];
+        
+        for(int i=0; i<dictionariesArray.count; i++)
+        {
+            makeHolder = [MTLJSONAdapter modelOfClass:[Make class] fromJSONDictionary:[dictionariesArray objectAtIndex:i] error:&parserError];
+            
+            if(parserError)
+            {
+                NSLog(@"%@",parserError);
+            }
+            
+            [self.makes addObject:makeHolder];
+            NSLog(@"%@",[(Make*)[self.makes objectAtIndex:i] nameAr]);
+            [indexPaths addObject:[NSIndexPath indexPathForItem:self.makes.count-1 inSection:0]];
+        }
+        
+        [self.makesCollectionView insertItemsAtIndexPaths:indexPaths];
+        
```

Comment:
you can make the parsing using this line of code 
 NSArray *makes = [MTLJSONAdapter modelsOfClass:[Make class] fromJSONArray:responseObject error:&error];
You don't need to create NsMutableArray for make and for indexpaths